include/psa/crypto_struct.h

```diff
@@ -158,16 +158,23 @@ struct psa_cipher_operation_s
     unsigned int key_set : 1;
     unsigned int iv_required : 1;
     unsigned int iv_set : 1;
+    unsigned int accelerator_set : 1;
```

Comment:
This is an ABI change (and that's what the pr-merge CI job is complaining about for e3fa23de388a8ea744dc9c5157968a0e85090660).

I wonder if `accelerator_set` should remain undefined if `MBEDTLS_PSA_CRYPTO_DRIVERS` is not defined. Technically, it's extra baggage that is unnecessary without `MBEDTLS_PSA_CRYPTO_DRIVERS`, but extra ifdefs would clutter the code.

Reply:
I'm very much in the camp of not guarding it, since it'll massively clutter `psa_crypto.c` with even more ifdef-guarding. If you have a halfway decent compiler with multifile compilation, it'll notice that when `MBEDTLS_PSA_CRYPTO_DRIVERS` is defined, all the wrapper calls return `NOT_SUPPORTED`, and so `accelerator_set` won't be touched. The only extra baggage that then remains is the explicit zeroing as part of `_abort` (which IMO could be replaced by a memset, but that's outside of my scope).

Reply:
`abort` needs to zeroize in case there's sensitive information (such as partially processed data) in the context.

Reply:
This comment may have been added to the wrong comment thread? I'm not arguing `abort` doesn't need to zeroize, just that it could do so using a `memset(operation, 0, sizeof(*operation))` instead of the current member-by-member zeroization.

Reply:
Suppose the application doesn't use the operation object anymore after the abort, and frees the memory, and the compiler notices this and optimizes the `memset` out (plausible with link-time optimization). Then the old contents of the operation object, which may include sensitive information, end up in unused memory, potentially vulnerable to a Heartbleed-type attack. This is precisely what we want to avoid with `zeroize`.